lib/engine/game/g_1817/game.rb

```diff
@@ -873,8 +873,9 @@ def rust(train)
         def remove_train(train)
           from_depot = train.from_depot?
           super
+
           return unless inventor_private&.owner&.corporation?
-          return if !from_depot || @depot.trains.find { |t| t.name == train.name } != train
+          return if train.rusted || !from_depot || @depot.trains.find { |t| t.name == train.name } != train
```

Comment:
I don't remember what this logic is attempting to do and I think it is hard to follow. I think we should go a simpler way of removing the value from @payouts after it pays out. That way, it'll only pay out once, like it should, and we don't have to track any additional state.

To do this, move the log message  and the return unless inventor_private&.owner&.corporation? to inventor_payout  and handle everything there. Then the return if rusted, !depot, etc. logic can be removed.

Reply:
Ah, made a commit that should implement this approach, check it out [5db9212](https://github.com/tobymao/18xx/pull/9011/commits/5db92125ab6b40c5c2969d8556d378e7330e3aec)

Reply:
Looks good, I think.